paasta_tools/spark_tools.py

```diff
@@ -234,11 +234,23 @@ def get_spark_driver_monitoring_annotations(
     """
     Returns Spark driver pod annotations - currently used for Prometheus metadata.
     """
-    ui_port_str = str(spark_config.get("spark.ui.port", ""))
-    annotations = {
-        "prometheus.io/port": ui_port_str,
-        "prometheus.io/path": "/metrics/prometheus",
-    }
+    annotations: Dict[str, str] = dict()
+
+    ui_port_str = spark_config.get("spark.ui.port", "")
+    if ui_port_str != "":
+        annotations["prometheus.io/port"] = ui_port_str
+        annotations["prometheus.io/path"] = "/metrics/prometheus"
+
+    paasta_service_non_truncated = spark_config.get(
+        "spark.kubernetes.executor.annotation.paasta.yelp.com/service", ""
+    )
+    paasta_instance_non_truncated = spark_config.get(
+        "spark.kubernetes.executor.annotation.paasta.yelp.com/instance", ""
+    )
+    if paasta_service_non_truncated != "" and paasta_instance_non_truncated != "":
+        annotations["paasta.yelp.com/service"] = paasta_service_non_truncated
+        annotations["paasta.yelp.com/instance"] = paasta_instance_non_truncated
```

Comment:
```suggestion
    annotations: Dict[str, str] = {}

    ui_port_str = spark_config.get("spark.ui.port")
    if ui_port_str:
        annotations.update(
            {
                "prometheus.io/port": ui_port_str,
                "prometheus.io/path": "/metrics/prometheus",
            }
        )
           
    paasta_service_non_truncated = spark_config.get(
        "spark.kubernetes.executor.annotation.paasta.yelp.com/service"
    )
    paasta_instance_non_truncated = spark_config.get(
        "spark.kubernetes.executor.annotation.paasta.yelp.com/instance"
    )
    if paasta_service_non_truncated and paasta_instance_non_truncated:
        annotations.update(
            {
                "paasta.yelp.com/service": paasta_service_non_truncated,
                "paasta.yelp.com/instance": paasta_instance_non_truncated,
            }
        )
```
i'd maybe do something like this, but the existing code works too (mostly personal preference around not using "" over None and preferring dict.update())